code/clue.js

```diff
@@ -1,59 +1,254 @@
+    }
+];
 
-const suspects = [
-  mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+//Array with weapons for shuffling favourite weapons.
+let weaponArray = [
+    'rope',
+    'knife',
+    'candlestick',
+    'dumbbell',
+    'poison',
+    'axe',
+    'bat',
+    'trophy',
+    'pistol'
+];
 
-const weapons = []
+//Shuffles the weapons and randomly sets new favourite wepons.
+const shufflefavouriteWeapons = (arrayOfSuspects) => {
+    arrayOfSuspects.forEach(suspect => {
+        suspect.weapon = weaponArray[Math.floor(Math.random() * 9)];
 
-const rooms = []
+    })
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+shufflefavouriteWeapons(suspects);
+
+
+// Randomly selects an item from the array you pass into the function
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+    return array[Math.floor(Math.random() * array.length)]
+}
+
+
+// An object with a killer, a favouriteWeapon and a room. The values will be set later.
+let mystery = {
+    killer: '',
+    weapon: '',
+    room: ''
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    // Randomly selects a killer from the suspects-array and adds that to the mystery object.
+    mystery.killer = randomSelector(suspects);
+    // This will change the background color of the card and show picture and detail about the suspect.
+    document.getElementById('killerCard').style.background = mystery.killer.color;
+    document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+    document.getElementById('killerImage').src = mystery.killer.image;
+    document.getElementById('killerAge').innerHTML = `${mystery.killer.age} years old`;
+    document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+    document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+}
+
+// This function will be invoked when you click on the favouriteWeapon card. 
+const pickWeapon = () => {
+    // Randomly selects a favouriteWeapon from the favouriteWeapons-array and adds that to the mystery object.
```

Comment:
Change this comment since you are picking a weapon for the weapon card but you are not using favouriteWeapons for the suspects.